### PR TITLE
implement http service

### PR DESCRIPTION
**Pull request by gozssky**

### CODE REVIEW

File: src/storage.rs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    where
+        F: FnMut(&mut State) -> bool,
+    {
+        if f(&mut *self.state.write().unwrap()) {
+            let state = self.dump();
+            let data = serde_json::to_vec(&state).unwrap();
+            let tmp_path = format!("{}.tmp", self.path);
+            tokio::fs::write(&tmp_path, data).await?;
+            tokio::fs::rename(&tmp_path, &self.path).await?;
+        }
```
Comment (gozssky) [inline src/storage.rs:117]: Need fix: lock must be held when dumping state to file.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.idea
```
Comment (hi-rustin) [inline .gitignore:1]:
```suggestion
.idea
.vscode
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+build:
+	cargo build
+
```
Comment (hi-rustin) [inline Makefile:3]:
```suggestion
clippy:
          cargo clippy
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+run:
+	cargo run
```
Comment (hi-rustin) [inline Makefile:5]: Do we need to lock the rust version? Add a toolchain file?

**Review by hi-rustin — APPROVED** [review]
LGTM

File: Makefile (hunk 4 above)
Comment (gozssky) [reply Makefile:6]: We can add it later.

**Pull request merged**